rules/TypeDeclaration/Rector/Class_/ReturnTypeFromStrictTernaryRector.php

```diff
@@ -95,7 +97,12 @@ public function refactorWithScope(Node $node, Scope $scope): ?Node
 
         $ternary = $return->expr;
 
-        $nativeTernaryType = $scope->getNativeType($ternary);
+        $returnScope = $return->expr->getAttribute(AttributeKey::SCOPE);
+        if ($returnScope === null) {
+            throw new ShouldNotHappenException();
```

Comment:
return null is better here, to avoid overlapped reprint node that scope may noy yet exists, that skipped at RectifiedAnalyzer.

Reply:
:+1:

done